ccb/upstream_project.py

```diff
@@ -249,6 +251,57 @@ def _get_owner_repo(cls, recipe):
         raise _Unsupported()
 
 
+class GithubReleaseProject(GithubProject):
```

Comment:
`UpstreamProject`s should be cheap to instantiate and do work lazily and asynchronously. Check out how `GitProject` is implemented: `__init__` does nothing and the logic is mostly in the method `versions` (which also caches its result).
I suggest you do the same:
- [ ] Move HTTP requests to `versions` (and cache the results)
- [ ] Use `aiohttp` instead of `requests` (see cci.py)

Reply:
With what you are suggesting,  you don't detect if there are actual releases and actual release tarball during construction, so there is no way to fallback on a different Upstream subclass. So it means that there must be a single class for git. I'll try to merge the classes tomorrow.